src/index.tsx

```diff
@@ -28,12 +28,21 @@ const INITIAL_DATA = {
   }
 };
 
-function getPositionX() {
-  return window.pageXOffset || 0;
+function hasOwnProp(obj: Object, prop: string) {
+  if (typeof obj === 'undefined' || obj === null) return false;
+  return Object.hasOwnProperty.call(obj, prop);
 }
 
-function getPositionY() {
-  return window.pageYOffset || 0;
+function getPositionX(elem: Window | HTMLElement) {
+  if (hasOwnProp(elem, 'pageXOffset')) return elem.pageXOffset;
+  if (hasOwnProp(elem, 'scrollLeft')) return elem.scrollLeft;
+  return 0
+}
```

Comment:
I don't have a lot of TS experience so I'm not really sure the correct way to type this. It's complaining about `pageXOffset` and `scrollLeft` not existing on `Window | HTMLElement`.